go/controller/internal/controller/toolserver_controller.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"github.com/kagent-dev/kagent/go/controller/internal/autogen"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	agentv1alpha1 "github.com/kagent-dev/kagent/go/controller/api/v1alpha1"
+)
+
+// ToolServerReconciler reconciles a ToolServer object
+type ToolServerReconciler struct {
+	client.Client
+	Scheme     *runtime.Scheme
+	Reconciler autogen.AutogenReconciler
+}
+
+// +kubebuilder:rbac:groups=agent.ai.solo.io,resources=toolservers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=agent.ai.solo.io,resources=toolservers/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=agent.ai.solo.io,resources=toolservers/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the ToolServer object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.20.0/pkg/reconcile
+func (r *ToolServerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	return ctrl.Result{
+		// loop forever because we need to refresh tools server status
```

Comment:
How does this work if the object is updated? Will this event continue to requeue forever